tests/lib/cli-engine.js

```diff
@@ -21,6 +21,10 @@ const assert = require("chai").assert,
 
 const proxyquire = require("proxyquire").noCallThru().noPreserveCache();
 
+// Need use 'flatted' to parse the cache content as the 'file-entry-cache' use it to create cache
+// eslint-disable-next-line node/no-extraneous-require
+const flatted = require("flatted");
```

Comment:
Why not add it as `devDependencies` to `package.json`?

Reply:
Because this `flatted` is not used directly, it's used by `file-entry-cache`. The code just use `file-entry-cache` to create the cache.

Reply:
This would cause potential problem, e.g. if `file-entry-cache` removes that dependency.

Reply:
Yes, that's the issue with the current test cases which need to use the same way as `file-entry-cache`

Reply:
Seems the tests should just use file-entry-cache directly in that case.

Reply:
Seems the tests should just use file-entry-cache directly in that case.

Reply:
yeah, that would be better, but `file-entry-cache` doesn't export any method about it.

Reply:
I share the same concern with @g-plane . And the tests passed depends on the package manager implements (not all the package managers have flatted the dependencies like npm v3+)

It's not ideal to add it to devdependency -- we have to make sure eslint is using the compatible version with `file-entry-cache`, that adding the maintaining burden. :-(

Reply:
@platinumazure @Teamop `file-entry-cache` has a method called `createFromFile` that can be used to read the cache from a given file the actual structure is loaded into the .cache property

Like this:

```js
const fCache = require('file-entry-cache');

const fileCache = fCache.createFromFile('/path/to/file');
const { cache } = fileCache; // cache has the actual structure loaded.
```

Reply:
thanks @royriojas , done